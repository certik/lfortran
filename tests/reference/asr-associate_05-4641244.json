{
    "basename": "asr-associate_05-4641244",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/associate_05.f90",
    "infile_hash": "f1a6dbfbf5fca41a24fc0c29a627f765ad77e22cf77713073c6b8029",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-associate_05-4641244.stdout",
    "stdout_hash": "6a5ad2d2dfafc588f27a3ea2b580fc3ab3c47c9e364b91fe7fe3d51f",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}