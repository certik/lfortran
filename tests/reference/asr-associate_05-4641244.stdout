(TranslationUnit (SymbolTable 1 {associate_05: (Program (SymbolTable 2 {a: (Variable 2 a Local () () Default (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) Source Public Required .false.), associate_block: (AssociateBlock (SymbolTable 3 {v: (Variable 3 v Local () () Allocatable (Pointer (Real 4 [(() ())])) Source Private Required .false.), z: (Variable 3 z Local () () Default (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) Source Private Required .false.)}) associate_block [(= (Var 3 z) (RealBinOp (RealUnaryMinus (RealBinOp (RealBinOp (Var 2 x) Mul (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToReal (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) (RealConstant   2.00000000000000000e+00 (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]))) (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) ()) Add (RealBinOp (Var 2 y) Mul (Cast (IntegerConstant 2 (Integer 4 [])) IntegerToReal (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) (RealConstant   2.00000000000000000e+00 (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]))) (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) ()) (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) ()) (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) ()) Mul (Var 2 theta) (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) ()) ()) (=> (Var 3 v) (Var 2 myreal)) (= (Var 3 v) (RealBinOp (Var 2 a) Add (Var 3 z) (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) ()) ()) (= (ArrayItem (Var 3 v) [(() (IntegerConstant 1 (Integer 4 [])) ())] (Real 4 []) ()) (RealBinOp (ArrayItem (Var 3 v) [(() (IntegerConstant 1 (Integer 4 [])) ())] (Real 4 []) ()) Mul (RealConstant   4.59999999999999964e+00 (Real 4 [])) (Real 4 []) ()) ()) (= (ArraySection (Var 3 v) [((IntegerConstant 2 (Integer 4 [])) (IntegerConstant 5 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))] (Real 4 [(() ())]) ()) (RealBinOp (ArraySection (Var 3 v) [((IntegerConstant 2 (Integer 4 [])) (IntegerConstant 5 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))] (Real 4 [(() ())]) ()) Mul (ArraySection (Var 3 v) [((IntegerConstant 2 (Integer 4 [])) (IntegerConstant 5 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))] (Real 4 [(() ())]) ()) (Real 4 [(() ())]) ()) ()) (= (ArraySection (Var 3 v) [((IntegerConstant 6 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))] (Real 4 [(() ())]) ()) (RealBinOp (ArraySection (Var 3 v) [((IntegerConstant 6 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])) (IntegerConstant 1 (Integer 4 [])))] (Real 4 [(() ())]) ()) Mul (RealConstant   2.60000000000000009e+00 (Real 4 [])) (Real 4 [(() ())]) ()) ())]), myreal: (Variable 2 myreal Local () () Allocatable (Real 4 [(() ())]) Source Public Required .false.), theta: (Variable 2 theta Local () () Default (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) Source Public Required .false.), x: (Variable 2 x Local () () Default (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) Source Public Required .false.), y: (Variable 2 y Local () () Default (Real 4 [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))]) Source Public Required .false.)}) associate_05 [] [(Allocate [(2 myreal [((IntegerConstant 1 (Integer 4 [])) (IntegerConstant 10 (Integer 4 [])))])] () () ()) (= (Var 2 x) (RealConstant   4.19999999999999984e-01 (Real 4 [])) ()) (= (Var 2 y) (RealConstant   3.49999999999999978e-01 (Real 4 [])) ()) (= (Var 2 myreal) (RealConstant   9.09999999999999964e+00 (Real 4 [])) ()) (= (Var 2 theta) (RealConstant   1.50000000000000000e+00 (Real 4 [])) ()) (= (Var 2 a) (RealConstant   4.00000000000000022e-01 (Real 4 [])) ()) (AssociateBlockCall 2 associate_block) (Print () [(Var 2 myreal)] () ()) (ImplicitDeallocate [2 myreal])])}) [])
